drivers/hid/hid-nintendo.c

```diff
@@ -1673,6 +1782,54 @@ static int joycon_input_create(struct joycon_ctlr *ctlr)
 			input_set_capability(ctlr->input, EV_KEY,
 					     joycon_button_inputs_r[i]);
 	}
+	if (jc_type_is_nescon(ctlr)) {
+ 		/* set up dpad hat */
+ 		input_set_abs_params(ctlr->input, ABS_HAT0X, -1, 1, 0, 0);
+ 		input_set_abs_params(ctlr->input, ABS_HAT0Y, -1, 1, 0, 0);
+
+ 		/* set up buttons */
+		for (i = 0; nescon_button_inputs[i] > 0; i++)
+ 			input_set_capability(ctlr->input, EV_KEY, nescon_button_inputs[i]);
+
+ 		/* register the device here, we don't need any more setup */
+ 		ret = input_register_device(ctlr->input);
+ 		if (ret)
+ 			return ret;
+
+ 		return 0;
+ 	}
```

Comment:
This conditional block, along with the following two, are highly duplicative. I split out these conditions from the changes I adapted. It seemed to me that the existing abstraction was starting to strain.

However, I may try to fold these again into one conditional in a future commit. For now, this works.